runtime/interpreter/value.go

```diff
@@ -1592,7 +1607,21 @@ func (v *ArrayValue) Transfer(
 	}
 
 	if isResourceKinded {
+		// Update the resource in-place,
+		// and also update all values that are referencing the same value
+		// (but currently point to an outdated Go instance of the value)
+
 		v.array = array
+
+		newStorageID := array.StorageID()
+		interpreter.updateReferencedResource(
+			currentStorageID,
+			newStorageID,
+			func(value ReferenceTrackedResourceKindedValue) {
+				value.(*ArrayValue).array = array
```

Comment:
Does this count as an unchecked cast? This wouldn't be picked up by the linter, but we are implicitly ignoring the boolean result here. Unsure whether or not it's worth doing the cast separately.

Reply:
Yes, it definitely does! Good point that we shouldn't introduce new type unchecked assertions like this one and the ones below 👍

Reply:
Fixed in 7295252fe948a5fa81f01fdd8c574234493a67a4